dotcom-rendering/src/lib/useSignedInAuthState.ts

```diff
@@ -0,0 +1,77 @@
+import type { IdentityAuthState } from '@guardian/identity-auth';
+import { useEffect, useState } from 'react';
+import type { CustomIdTokenClaims } from './identity';
+
+export type AuthStateStatus = 'Pending' | 'NotInTest' | 'Ready';
+
+export const getOptionsHeadersWithOkta = (
+	status: AuthStateStatus,
+	state: IdentityAuthState,
+): RequestInit => {
+	if (status === 'NotInTest') {
+		return {
+			credentials: 'include',
+		};
+	}
+
+	if (status === 'Ready' && state.accessToken?.accessToken) {
+		return {
+			headers: {
+				Authorization: `Bearer ${state.accessToken.accessToken}`,
+			},
+		};
+	}
+
+	return {};
+};
+
+export async function getAuthState(): Promise<
+	IdentityAuthState<never, CustomIdTokenClaims>
+> {
+	const { isSignInWithOktaAuthState } = await import('./identity');
+	const authState = await isSignInWithOktaAuthState();
+	return authState;
+}
+
+export async function eitherSignedInWithOktaOrElse<A, B>(
+	inOktaTestFunction: (
+		authState: IdentityAuthState<never, CustomIdTokenClaims>,
+	) => A,
+	notInOktaTestFunction: () => B,
+): Promise<A | B> {
+	const isInOktaExperiment =
+		window.guardian.config.tests.oktaVariant === 'variant';
+
+	if (isInOktaExperiment) {
+		const authState = await getAuthState();
+		return inOktaTestFunction(authState);
+	} else {
+		return notInOktaTestFunction();
+	}
+}
+
+export const useSignedInAuthState = (): [
+	AuthStateStatus,
+	IdentityAuthState<never, CustomIdTokenClaims>,
+] => {
+	const [status, setStatus] = useState<AuthStateStatus>('Pending');
```

Comment:
Since this is of type `AuthStateStatus`, maybe this should be
```suggestion
	const [authStateStatus, setAuthStateStatus] = useState<AuthStateStatus>('Pending');
```